code/modules/projectiles/guns/ballistic/rifle.dm

```diff
@@ -28,6 +30,14 @@
 	drop_bolt(user)
 
 
+/obj/item/gun/ballistic/rifle/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	if(need_bolt_lock_to_interact)
+		if(!bolt_locked && !istype(tool, /obj/item/knife))
+			balloon_alert(user, "bolt closed!")
+			return
```

Comment:
```suggestion
	if(need_bolt_lock_to_interact && !bolt_locked && !istype(tool, /obj/item/knife))
		balloon_alert(user, "bolt closed!")
		return
```
By the by, are you sure it wouldn't be better to blacklist interactions with ammo/mags when the bolt is closed instead? This may cause some issues in the future.